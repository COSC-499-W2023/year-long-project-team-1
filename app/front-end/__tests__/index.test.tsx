/*
 * Created on Sun Oct 22 2023
 * Author: Connor Doman
 */

import { render, screen } from "@testing-library/react";
import Home from "../src/app/page";
import "@testing-library/jest-dom";

/* Example Test: Does the page at `/` render a header? (Should FAIL) */
describe("Home Page", () => {
    it("page has 'Test Login' render properly semantic heading", () => {
        render(<Home />);
        expect(screen.getByRole("heading", { name: "Test Login" })).toBeInTheDocument();
    });

    it("homepage is unchanged from snapshot", () => {
        const { container } = render(<Home />);
        expect(container).toMatchSnapshot();
    });
});
