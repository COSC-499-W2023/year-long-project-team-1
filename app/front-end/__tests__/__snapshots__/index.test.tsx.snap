// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page homepage is unchanged from snapshot 1`] = `
<div>
  <main
    class="column"
  >
    <h2
      class="subheading"
    >
      Test Login
    </h2>
    <p>
      You can login with the username "
      <code>
        johnny@example.com
      </code>
      " and password "
      <code>
        password
      </code>
      "
    </p>
    <br />
    <p>
      You can visit 
      <a
        href="/api/session"
      >
        this link
      </a>
       to see a breakdown of your current session.
    </p>
  </main>
</div>
`;
